tests/test_nn/test_modules.py

```diff
@@ -118,6 +118,38 @@ def _exp_score(self, h, r, t, mlp) -> torch.FloatTensor:  # noqa: D102
         return mlp(x).view(1, -1) @ t.view(-1, 1)
 
 
+class HAKETests(cases.InteractionTestCase):
+    """Tests for HAKE interaction."""
+
+    cls = pykeen.nn.modules.HAKEInteraction
+
+    def _exp_score(
+        self,
+        h_phase: torch.FloatTensor,
+        h_modulus: torch.FloatTensor,
+        r_phase: torch.FloatTensor,
+        r_modulus: torch.FloatTensor,
+        r_bias: torch.FloatTensor,
+        t_phase: torch.FloatTensor,
+        t_modulus: torch.FloatTensor,
+        gamma: float,
+        modulus_weight: float,
+        phase_weight: float,
+    ) -> torch.FloatTensor:  # noqa: D102
+        # remove batch/num dimension
+        h_phase, h_modulus, r_phase, r_modulus, r_bias, t_phase, t_modulus = strip_dim(
+            h_phase, h_modulus, r_phase, r_modulus, r_bias, t_phase, t_modulus)
+        r_bias = r_bias.clamp(max=1)
```

Comment:
This is copy-paste from the interaction function. Is there a better way to do this?

Reply:
If there isn't a more clever way to test the scoring works properly without re-writing the interaction function itself, this test isn't super useful anyway :/

Reply:
It checks against broadcasting issues (although this could also be done directly on the functional form).

Reply:
Although it only looks at the case of tensor of shape `(1,1,1,1,*d)` vs. `(*d)` :thinking: